src/index.js

```diff
@@ -112,6 +113,29 @@ class MongoOplogReader {
     // re-register this worker every 10 seconds by default
     const healthcheckMs = this.healthcheckDuration * 1000;
     this.healthcheckInterval = setInterval(() => this.registerWorker(), healthcheckMs);
+
+    // adjust throttling every 30 seconds
+    this.throttleAdjustmentInterval = setInterval(() => {
+      this.redisClient.infoAsync('memory')
+        .then(info => {
+          const usedMem = info.match(/used_memory:(\d*)/)[1];
+          const totalMem = info.match(/total_system_memory:(\d*)/)[1];
+          const memUsage = usedMem / totalMem;
+          /**
+           * Throttle the throughput based on redis memory usage. For example, assuming 24 hour TTL:
+           * Redis usage:  |  Delay:       
+           *           0%  |  none
+           *          50%  |  none
+           *          75%  |  none
+           *        75.1%  |  ~5 minutes
+           *          76%  |  ~1 hour
+           *         100%  |  ~24 hours
+           */
+          this.throttleDelay = this.ttl * Math.max(memUsage - 0.75, 0) * 4;
```

Comment:
is `this.ttl` in ms/sec/min and is `4` just an arbitrary number?

Reply:
`ttl` is in seconds. I picked 75% arbitrarily. The `4` is actually the same as `1 / (1 - 0.75)` so I just updated the formula so it doesn't appear arbitrary.

Reply:
cool, makes sense!